src/core/drive/navigator.js

```diff
@@ -125,6 +125,7 @@ export class Navigator {
 
   visitCompleted(visit) {
     this.delegate.visitCompleted(visit)
+    this.stop()
```

Comment:
I see that, before, it was invoking `stop` only when starting visits and submitting forms. This looks correct but, since this is a pretty important path, any concerns about introducing unwanted side effects here? I can't think of anything in particular, but concerned about interrupting form submissions we were not interrupting before.

Reply:
> This looks correct but, since this is a pretty important path, any concerns about introducing unwanted side effects here?

Now that you mention it, a `Visit` initiated by a link click is guaranteed to either do nothing (when canceled) or navigated. A form, on the other hand, has the potential to complete and succeed without navigating when handling stream responses. The same is true for a link click with `[data-turbo-stream]`.

Acknowledging that, what *is* the desired order of operations for a Page Refresh that's competing with a Stream form submission?

It feels like the Visit should always win out, since that's requesting the most (and most up-to-date) content. Is that correct?

Reply:
@jorgemanrubia I've pushed up a change to remove the call to `this.stop()`, and replace it with `delete this.currentVisit`.

Reply:
@jorgemanrubia are you available to re-review this?